discovery-infra/test_infra/helper_classes/proxy_params.py

```diff
@@ -0,0 +1,29 @@
+from test_infra.helper_classes.idict import IDict
+
+
+class Proxy(IDict):
+    """Proxy settings for the installation.
+
+    Args:
+        http_proxy (str): endpoint for accessing in every HTTP request.
+        https_proxy (str): endpoint for accessing in every HTTPS request.
+        no_proxy (str): comma separated values of addresses/address ranges/DNS entries
+            that shouldn't be accessed via proxies.
+    """
+
+    def __init__(
+        self,
+        http_proxy: str,
+        https_proxy: str,
+        no_proxy: str,
+    ):
+        self.http_proxy = http_proxy
+        self.https_proxy = https_proxy
+        self.no_proxy = no_proxy
+
+    def as_dict(self) -> dict:
+        return {
+            "httpProxy": self.http_proxy,
+            "httpsProxy": self.https_proxy,
+            "noProxy": self.no_proxy,
+        }
```

Comment:
nit: can you add a new line at the end of the file?
on Linux sometimes pipes have weird results when files don't have new lines at the bottom